src/lib/core/components/visualizations/VisualizationPlugin.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  IsEnabledInPickerParams,
+  VisualizationProps,
+} from './VisualizationTypes';
+
+export interface VisualizationPluginSpec<Options = undefined> {
+  /** Component used to render the visualization */
+  fullscreenComponent: React.ComponentType<VisualizationProps<Options>>;
+  /** Image URI used in selector */
+  selectorIcon: string;
+  /** Function used to create a default configuration when a visualization instance is created */
+  createDefaultConfig: () => unknown;
+  /** Function used to determine if visualization is compatible with study */
+  isEnabledInPicker?: (props: IsEnabledInPickerParams) => boolean;
+}
+
+export interface VisualizationPlugin<Options = undefined>
+  extends VisualizationPluginSpec<Options> {
+  /** Options that have been set for the plugin instance */
+  options?: Options;
+  /** Factory function to create a new plugin instance with options applied */
+  withOptions: (options: Options) => VisualizationPlugin<Options>;
+  /** Factory function to create a new plugin instance with a new selector icon */
+  withSelectorIcon: (icon: string) => VisualizationPlugin<Options>;
```

Comment:
Successfully used to get the new selector icon to show up! However once i click on the icon i end up in a big type error in `InputVariables.tsx` somehow. Maybe you've seen this before?

Reply:
I will take a look!

Reply:
as long as it works for you that's okay with me - i may have been implementing it incorrectly!

Reply:
fixed with https://github.com/VEuPathDB/web-eda/pull/1244/commits/49f428a4e16fe64b779afc98b5d332a5d490bac3